extensions/jms/src/main/java/org/mule/extensions/jms/api/operation/JmsRequestReply.java

```diff
@@ -0,0 +1,278 @@
+   * @param deliveryDelayUnit Time unit to be used in the deliveryDelay configurations
+   * @return a {@link Result} with the reply {@link Message} content as {@link Result#getOutput} and its properties
+   * and headers as {@link Result#getAttributes}
+   * @throws JmsExtensionException if an error occurs
+   */
+  @OutputResolver(output = JmsOutputResolver.class)
+  public Result<Object, JmsAttributes> requestReply(@UseConfig JmsProducerConfig config, @Connection JmsConnection connection,
+                                                    @XmlHints(
+                                                        allowReferences = false) @Summary("The name of the Destination where the Message should be sent") String destination,
+                                                    @Optional(
+                                                        defaultValue = "false") @Summary("The kind of the Destination") boolean isTopic,
+                                                    @Summary("A builder for the message that will be published") MessageBuilder messageBuilder,
+                                                    @Optional @Summary("If true, the Message will be sent using the PERSISTENT JMSDeliveryMode") Boolean persistentDelivery,
+                                                    @Optional @Summary("The default JMSPriority value to be used when sending the message") Integer priority,
+                                                    @Optional @Summary("Defines the default time the message will be in the broker before it expires and is discarded") Long timeToLive,
+                                                    @Optional @Summary("Time unit to be used in the timeToLive configurations") TimeUnit timeToLiveUnit,
+                                                    @Optional(
+                                                        defaultValue = "10000") @Summary("Maximum time to wait for a response") Long maximumWaitTime,
+                                                    @Optional String encoding,
+                                                    @Optional AckMode ackMode,
+                                                    // JMS 2.0
+                                                    @Optional @Summary("Only used by JMS 2.0. Sets the delivery delay to be applied in order to postpone the Message delivery") Long deliveryDelay,
+                                                    @Optional @Summary("Time unit to be used in the deliveryDelay configurations") TimeUnit deliveryDelayUnit)
+      throws JmsExtensionException {
+
+    JmsSession session = null;
+    MessageProducer producer = null;
+    MessageConsumer consumer = null;
+
+    java.util.Optional<Long> delay = resolveDeliveryDelay(connection.getJmsSupport().getSpecification(),
+                                                          config, deliveryDelay, deliveryDelayUnit);
+
+    try {
+      // TODO refactor this with JmsPublish and JmsConsume operation
```

Comment:
request-reply is mostly the same code than the publish for request and than consume for listening the reply. It would be nice to have reusable clients or commands for this